File: android/app/src/main/java/com/plantora/billing/domain/Product.kt
package com.plantora.billing.domain

data class Product(
    val id: String,
    val name: String,
    val category: String?,
    val retailPrice: Money,
    /** Absolute, ready-to-load image URL (or null). */
    val photoUrl: String?,
    val isActive: Boolean,
)

File: Plantbill/Plantbill/Features/Products/Product.swift
import Foundation

/// Mirrors backend `ProductOut` (app/schemas/product.py).
struct Product: Decodable, Identifiable, Equatable {
    let id: UUID
    let name: String
    let category: String?
    let retailPrice: String
    let lastWholesalePrice: String?
    let photoUrl: String?
    let isActive: Bool
    let createdAt: Date

    var price: Money { Money.parse(retailPrice) }
    var resolvedPhotoURL: URL? { MediaURL.resolve(photoUrl) }
}
